### DISCUSSION TITLE
What text normalizers are out there for abbreviations, units, roman numerals, etc?

### DISCUSSION THREAD

**Discussion by xenotropic** [coqui-ai/TTS#2443] [General Q&A]
What are people using to normalize text prior to rendering to speech, to do things like expand "Mr." to "Mister", 12 ha to "twelve hectares" numbers from "1976" to "nineteen seventy six", and will make "Charles V" into "Charles the fifth" but not "Malcolm X" into "Malcolm the tenth"? I can see there is cleaners.py in this repo, which is pretty basic; and then there is Nvidia's Nemo, which sort of does the job but not all of it. Are there any other options on the menu for this task?

**Comment by p0p4k** [comment]
https://github.com/search?q=text+normalization

**Comment by xenotropic** [comment]
I'm aware of how to do a search on github, thanks, and I did one before asking. Most of those results are for Chinese, or NLP, or for a Kaggle and can't easily be used elsewhere, etc. None on the first three pages of results at least can do SI units or roman numerals for STT, which seems pretty basic. Hence the question "what are people using" not "what repos exist".
